File: CyFile-Android/app/src/test/java/at/tugraz/tc/cyfile/secret/SimplePinPatternSecretVerifierUnitTest.java
package at.tugraz.tc.cyfile.secret;

import junit.framework.Assert;

import org.junit.Before;
import org.junit.Test;
import org.mockito.Mock;
import org.mockito.MockitoAnnotations;

import static org.mockito.Mockito.atLeastOnce;
import static org.mockito.Mockito.verify;
import static org.mockito.Mockito.when;

public class SimplePinPatternSecretVerifierUnitTest {

    private SimplePinPatternSecretVerifier simplePinPatternSecretVerifier;

    @Mock
    private SecretRepository secretRepository;

    @Before
    public void setup() {
        MockitoAnnotations.initMocks(this);
        simplePinPatternSecretVerifier = new SimplePinPatternSecretVerifier(secretRepository);
    }


    @Test
    public void verifyShouldReturnTrueIfTheValuesAreEqual() {
        when(secretRepository.getSecret())
                .thenReturn(() -> "test");

        boolean isValid = simplePinPatternSecretVerifier.verify(() -> "test");

        Assert.assertTrue(isValid);
        verify(secretRepository, atLeastOnce()).getSecret();
    }

    @Test
    public void verifyShouldReturnFalseWhenValuesAreNotEqual() {
        when(secretRepository.getSecret())
                .thenReturn(() -> "test");

        boolean isValid = simplePinPatternSecretVerifier.verify(() -> "invalidpin");

        Assert.assertFalse(isValid);
        verify(secretRepository, atLeastOnce()).getSecret();
    }

    @Test
    public void verifyShouldReturnFalseWhenNullPassed() {
        when(secretRepository.getSecret())
                .thenReturn(() -> "test");

        boolean isValid = simplePinPatternSecretVerifier.verify(null);

        Assert.assertFalse(isValid);
        verify(secretRepository, atLeastOnce()).getSecret();
    }

    @Test
    public void verifyShouldReturnFalseWhenSecretNotSet() {
        when(secretRepository.getSecret())
                .thenReturn(null);

        boolean isValid = simplePinPatternSecretVerifier.verify(() -> "test");

        Assert.assertFalse(isValid);
        verify(secretRepository, atLeastOnce()).getSecret();
    }
}

File: CyFile-Android/app/src/test/java/at/tugraz/tc/cyfile/secret/SimplePinPatternSecretVerifierUnitTest.kt
package at.tugraz.tc.cyfile.secret

import at.tugraz.tc.cyfile.BaseUnitTest
import at.tugraz.tc.cyfile.secret.impl.PinPatternSecret
import at.tugraz.tc.cyfile.secret.impl.SimplePinPatternSecretVerifier
import io.mockk.every
import io.mockk.impl.annotations.MockK
import io.mockk.verify
import junit.framework.Assert
import org.junit.Before
import org.junit.Test

class SimplePinPatternSecretVerifierUnitTest : BaseUnitTest() {

    private var simplePinPatternSecretVerifier: SimplePinPatternSecretVerifier? = null

    @MockK
    private lateinit var secretRepository: SecretRepository

    @Before
    fun setup() {
        simplePinPatternSecretVerifier = SimplePinPatternSecretVerifier(secretRepository)
    }


    @Test
    fun verifyShouldReturnTrueIfTheValuesAreEqual() {
        every { secretRepository.retrieveSecret() } returns PinPatternSecret("test")

        val isValid = simplePinPatternSecretVerifier!!.verify(PinPatternSecret("test"))

        Assert.assertTrue(isValid)

        verify { secretRepository.retrieveSecret() }
    }

    @Test
    fun verifyShouldReturnFalseWhenValuesAreNotEqual() {
        every { secretRepository.retrieveSecret() } returns PinPatternSecret("test")

        val isValid = simplePinPatternSecretVerifier!!.verify(PinPatternSecret("invalidpin"))

        Assert.assertFalse(isValid)

        verify { secretRepository.retrieveSecret() }
    }

    @Test
    fun verifyShouldReturnFalseWhenSecretNotSet() {
        every { secretRepository.retrieveSecret() } returns null

        val isValid = simplePinPatternSecretVerifier!!.verify(PinPatternSecret("test"))

        Assert.assertFalse(isValid)
        verify { secretRepository.retrieveSecret() }
    }
}
